examples/redux-saga-typescript-graphql-fluentui/pages/_app.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import App, { AppInitialProps, AppContext } from 'next/app'
+import { END } from 'redux-saga'
+import { wrapper, SagaStore } from '../store'
+
+import 'components/layouts/layout.css'
+
+class WrappedApp extends App<AppInitialProps> {
+  public static getInitialProps = async ({ Component, ctx }: AppContext) => {
+    // 1. Wait for all page actions to dispatch
+    const pageProps = {
+      ...(Component.getInitialProps
+        ? await Component.getInitialProps(ctx)
+        : {}),
+    }
+
+    // 2. Stop the saga if on server
+    if (ctx.req) {
+      ctx.store.dispatch(END)
+      await (ctx.store as SagaStore).sagaTask?.toPromise()
+    }
+
+    // 3. Return props
+    return {
+      pageProps,
+    }
+  }
```

Comment:
We are currently not taking examples using `getInitialProps` for simple stores, take a look at our other Redux examples to see how it works without it, e.g: https://github.com/vercel/next.js/tree/canary/examples/with-redux-saga